runtime/syntax/qf.vim

```diff
@@ -19,7 +19,7 @@ syn match	qfError		"error"	 contained
 syn cluster	qfType	contains=qfError
 
 " Hide file name and line number for help outline (TOC).
-if has_key(w:, 'qf_toc') || get(w:, 'quickfix_title') =~# '\<TOC$'
+if has_key(w:, 'qf_toc') || get(w:, 'quickfix_title') =~# '\<TOC$\|\<Table of contents\>'
```

Comment:
idk about this but since you're just augmenting an existing thing, we can just move along... :D